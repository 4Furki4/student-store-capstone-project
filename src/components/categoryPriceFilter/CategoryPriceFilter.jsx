/* eslint-disable unused-imports/no-unused-vars */
import { useTranslation } from "next-i18next";
import React from "react";

import Filter from "../filter";
import categories from "../filter/categories";
import PriceFilter from "../price-filter";

const CategoryPriceFilter = ({
    products,
    filteredProducts,
    setfilteredProducts,
    maxPrice,
    setMaxPrice,
    minPrice,
    setMinPrice,
    currentCategory,
    setCurrentCategory,
    productName,
    setProductName,
}) => {
    function categoriesFilter(category) {
        setfilteredProducts(
            products.filter((product) => product.category === category)
        );
        setProductName("");
        setMaxPrice("");
        setMinPrice("");
    }

    const { t } = useTranslation("category-price-filter");
    return (
        <div className='px-[15px] pb-4 xxs:mt-[-3.5rem] md:pt-5 lg:mt-0'>
            <div className='invisible mx-5 my-2 grid gap-5 text-2xl font-bold text-purple lg:visible lg:grid-cols-3'>
                <div className='col-span-2'>
                    <h1 className='brush w-fit'>{t("categories")}</h1>
                </div>
                <div>
                    <h1 className='brush w-fit'>{t("price")}</h1>
                </div>
            </div>
            <div className='grid overflow-hidden rounded-full bg-white xxs:grid-cols-1 sm:grid-cols-1 md:grid-cols-1 lg:grid-cols-3'>
                <div className='col-span-2 my-3 grid grid-flow-col justify-center overflow-x-auto scrollbar-hide xxs:gap-0 xxs:pt-4 md:gap-6 md:pt-2'>
                    {categories.map((catergory, i) => (
                        <Filter
                            key={catergory[i].id}
                            bgColor={catergory[i].color}
                            svg={catergory[i].svg}
                            name={catergory[i].name}
                            id={catergory[i].id}
                            categoriesFilter={categoriesFilter}
                            currentCategory={currentCategory}
                            setCurrentCategory={setCurrentCategory}
                        />
                    ))}
                </div>
                <div className='my-3 flex items-center justify-center'>
                    <PriceFilter
                        products={products}
                        filteredProducts={filteredProducts}
                        setfilteredProducts={setfilteredProducts}
                        maxPrice={maxPrice}
                        setMaxPrice={setMaxPrice}
                        minPrice={minPrice}
                        setMinPrice={setMinPrice}
                        currentCategory={currentCategory}
                        setCurrentCategory={setCurrentCategory}
                    />
                </div>
            </div>
        </div>
    );
};

export default CategoryPriceFilter;
