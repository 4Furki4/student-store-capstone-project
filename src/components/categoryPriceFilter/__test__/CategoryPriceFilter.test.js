import renderer from "react-test-renderer";

import CategoryPriceFilter from "../CategoryPriceFilter";

it("renders correctly", () => {
    const tree = renderer.create(<CategoryPriceFilter />).toJSON();
    expect(tree).toMatchSnapshot();
});
