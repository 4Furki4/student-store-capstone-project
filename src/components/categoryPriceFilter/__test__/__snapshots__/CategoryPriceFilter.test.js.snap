// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-[15px] pb-4 xxs:mt-[-3.5rem] md:pt-5 lg:mt-0"
>
  <div
    className="invisible mx-5 my-2 grid gap-5 text-2xl font-bold text-purple lg:visible lg:grid-cols-3"
  >
    <div
      className="col-span-2"
    >
      <h1
        className="brush w-fit"
      >
        categories
      </h1>
    </div>
    <div>
      <h1
        className="brush w-fit"
      >
        price
      </h1>
    </div>
  </div>
  <div
    className="grid overflow-hidden rounded-full bg-white xxs:grid-cols-1 sm:grid-cols-1 md:grid-cols-1 lg:grid-cols-3"
  >
    <div
      className="col-span-2 my-3 grid grid-flow-col justify-center overflow-x-auto scrollbar-hide xxs:gap-0 xxs:pt-4 md:gap-6 md:pt-2"
    >
      <div
        className="xxs:ml-[14rem] ne:ml-0 lg:ml-[8rem] xl:ml-0 xs:mr-0 scale-90 xxs:scale-90 lg:scale-75 xl:scale-90"
        onClick={[Function]}
      >
        <div
          className="flashing flex h-20 w-20 items-center justify-center rounded-full hover:cursor-pointer xxs:scale-[50%] sm:scale-90 md:scale-90 lg:scale-100 bg-purple"
        >
          <svg
            className="h-12 w-12 shrink fill-white"
            viewBox="0 0 24 24"
          >
            <path
              className="text-center"
              clipRule="evenodd"
              d="M2 6s1.5-2 5-2s5 2 5 2v14s-1.5-1-5-1s-5 1-5 1V6Zm10 0s1.5-2 5-2s5 2 5 2v14s-1.5-1-5-1s-5 1-5 1V6Z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h1
          className="text-center font-sans text-extraDarkPurple xxs:mt-[-18px] xxs:scale-75 xxs:pb-2 sm:mt-0 md:scale-90 md:pb-0 lg:pt-2"
        >
          Books
        </h1>
      </div>
      <div
        className="xs:ml-0 xs:mr-0 scale-90 xxs:scale-90 lg:scale-75 xl:scale-90"
        onClick={[Function]}
      >
        <div
          className="flashing flex h-20 w-20 items-center justify-center rounded-full hover:cursor-pointer xxs:scale-[50%] sm:scale-90 md:scale-90 lg:scale-100 bg-orange"
        >
          <svg
            className="h-12 w-12 shrink fill-white"
            viewBox="0 0 24 24"
          >
            <path
              className="text-center"
              clipRule="evenodd"
              d="M20 8V6c0-1.65-1.35-3-3-3H7C5.35 3 4 4.35 4 6v2c-1.65 0-3 1.35-3 3v5c0 1.65 1.35 3 3 3v1c0 .55.45 1 1 1s1-.45 1-1v-1h12v1c0 .55.45 1 1 1s1-.45 1-1v-1c1.65 0 3-1.35 3-3v-5c0-1.65-1.35-3-3-3zM6 6c0-.55.45-1 1-1h10c.55 0 1 .45 1 1v2.78c-.61.55-1 1.34-1 2.22v2H7v-2c0-.88-.39-1.67-1-2.22V6zm15 10c0 .55-.45 1-1 1H4c-.55 0-1-.45-1-1v-5c0-.55.45-1 1-1s1 .45 1 1v4h14v-4c0-.55.45-1 1-1s1 .45 1 1v5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h1
          className="text-center font-sans text-extraDarkPurple xxs:mt-[-18px] xxs:scale-75 xxs:pb-2 sm:mt-0 md:scale-90 md:pb-0 lg:pt-2"
        >
          Furniture
        </h1>
      </div>
      <div
        className="xs:ml-0 xs:mr-0 scale-90 xxs:scale-90 lg:scale-75 xl:scale-90"
        onClick={[Function]}
      >
        <div
          className="flashing flex h-20 w-20 items-center justify-center rounded-full hover:cursor-pointer xxs:scale-[50%] sm:scale-90 md:scale-90 lg:scale-100 bg-cyan"
        >
          <svg
            className="h-12 w-12 shrink fill-white"
            viewBox="0 0 24 24"
          >
            <path
              className="text-center"
              clipRule="evenodd"
              d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H1c-.55 0-1 .45-1 1s.45 1 1 1h22c.55 0 1-.45 1-1s-.45-1-1-1h-3zM5 6h14c.55 0 1 .45 1 1v8c0 .55-.45 1-1 1H5c-.55 0-1-.45-1-1V7c0-.55.45-1 1-1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h1
          className="text-center font-sans text-extraDarkPurple xxs:mt-[-18px] xxs:scale-75 xxs:pb-2 sm:mt-0 md:scale-90 md:pb-0 lg:pt-2"
        >
          Electronics
        </h1>
      </div>
      <div
        className="xs:ml-0 xs:mr-0 scale-90 xxs:scale-90 lg:scale-75 xl:scale-90"
        onClick={[Function]}
      >
        <div
          className="flashing flex h-20 w-20 items-center justify-center rounded-full hover:cursor-pointer xxs:scale-[50%] sm:scale-90 md:scale-90 lg:scale-100 bg-extraDarkPurple"
        >
          <svg
            className="h-12 w-12 shrink fill-white"
            viewBox="0 0 24 24"
          >
            <path
              className="text-center"
              clipRule="evenodd"
              d="M15 4l6 2v5h-3v8a1 1 0 0 1 -1 1h-10a1 1 0 0 1 -1 -1v-8h-3v-5l6 -2a3 3 0 0 0 6 0"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h1
          className="text-center font-sans text-extraDarkPurple xxs:mt-[-18px] xxs:scale-75 xxs:pb-2 sm:mt-0 md:scale-90 md:pb-0 lg:pt-2"
        >
          Clothes
        </h1>
      </div>
      <div
        className="xs:ml-0 xs:mr-0 scale-90 xxs:scale-90 lg:scale-75 xl:scale-90"
        onClick={[Function]}
      >
        <div
          className="flashing flex h-20 w-20 items-center justify-center rounded-full hover:cursor-pointer xxs:scale-[50%] sm:scale-90 md:scale-90 lg:scale-100 bg-yellow"
        >
          <svg
            className="h-12 w-12 shrink fill-white"
            viewBox="0 0 24 24"
          >
            <path
              className="text-center"
              clipRule="evenodd"
              d="M 22.238281 1.761719 C 21.085938 0.613281 19.449219 0.00390625 17.496094 0.00390625 C 13.75 0.00390625 9.363281 2.15625 5.757812 5.757812 C 3.035156 8.480469 1.117188 11.6875 0.351562 14.78125 C -0.417969 17.910156 0.0820312 20.558594 1.761719 22.238281 C 2.914062 23.386719 4.550781 23.996094 6.503906 23.996094 C 10.25 23.996094 14.636719 21.84375 18.242188 18.242188 C 20.964844 15.519531 22.882812 12.3125 23.648438 9.21875 C 24.417969 6.089844 23.917969 3.441406 22.238281 1.761719 Z M 9.523438 19.769531 C 9.488281 19.804688 9.441406 19.824219 9.390625 19.824219 C 9.386719 19.824219 9.378906 19.824219 9.375 19.824219 C 8.371094 19.730469 7.128906 19.027344 6.050781 17.949219 C 4.972656 16.871094 4.269531 15.628906 4.175781 14.625 C 4.171875 14.570312 4.191406 14.515625 4.230469 14.476562 C 4.269531 14.4375 4.324219 14.414062 4.378906 14.421875 C 5.382812 14.515625 6.628906 15.214844 7.707031 16.296875 C 8.785156 17.375 9.484375 18.617188 9.578125 19.621094 C 9.585938 19.675781 9.566406 19.730469 9.523438 19.769531 Z M 16.117188 16.117188 C 16.082031 16.152344 16.035156 16.171875 15.984375 16.171875 C 15.976562 16.171875 15.972656 16.171875 15.964844 16.171875 C 14.425781 16.023438 12.433594 14.898438 10.769531 13.230469 C 9.101562 11.566406 7.976562 9.574219 7.828125 8.035156 C 7.824219 7.976562 7.84375 7.921875 7.882812 7.882812 C 7.921875 7.84375 7.976562 7.824219 8.035156 7.828125 C 9.574219 7.976562 11.566406 9.101562 13.230469 10.769531 C 14.898438 12.433594 16.023438 14.425781 16.171875 15.964844 C 16.175781 16.023438 16.15625 16.078125 16.117188 16.117188 Z M 19.769531 9.523438 C 19.734375 9.558594 19.6875 9.578125 19.636719 9.578125 C 19.632812 9.578125 19.625 9.578125 19.621094 9.578125 C 18.617188 9.484375 17.375 8.785156 16.296875 7.707031 C 15.214844 6.625 14.515625 5.382812 14.421875 4.378906 C 14.414062 4.324219 14.4375 4.269531 14.476562 4.230469 C 14.515625 4.191406 14.570312 4.171875 14.625 4.175781 C 15.628906 4.269531 16.871094 4.972656 17.949219 6.050781 C 19.027344 7.128906 19.730469 8.371094 19.824219 9.375 C 19.828125 9.429688 19.808594 9.484375 19.769531 9.523438 Z M 19.769531 9.523438"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h1
          className="text-center font-sans text-extraDarkPurple xxs:mt-[-18px] xxs:scale-75 xxs:pb-2 sm:mt-0 md:scale-90 md:pb-0 lg:pt-2"
        >
          Food
        </h1>
      </div>
      <div
        className="xs:ml-0 xxs:mr-[2rem] ne:mr-0 lg:mr-[2rem] xl:mr-0 scale-90 xxs:scale-90 lg:scale-75 xl:scale-90"
        onClick={[Function]}
      >
        <div
          className="flashing flex h-20 w-20 items-center justify-center rounded-full hover:cursor-pointer xxs:scale-[50%] sm:scale-90 md:scale-90 lg:scale-100 bg-hadrisblue"
        >
          <svg
            className="h-12 w-12 shrink fill-white"
            viewBox="0 0 24 24"
          >
            <path
              className="text-center"
              clipRule="evenodd"
              d="M7.82 19H15v-1c0-2.21 1.79-4 4-4h.74l-1.9-8.44A2.009 2.009 0 0 0 15.89 4H12v2h3.89l1.4 6.25h-.01A6.008 6.008 0 0 0 13.09 17H7.82a2.996 2.996 0 0 0-3.42-1.94c-1.18.23-2.13 1.2-2.35 2.38A3.002 3.002 0 0 0 5 21c1.3 0 2.4-.84 2.82-2M5 19c-.55 0-1-.45-1-1s.45-1 1-1s1 .45 1 1s-.45 1-1 1m14-4c-1.66 0-3 1.34-3 3s1.34 3 3 3s3-1.34 3-3s-1.34-3-3-3m0 4c-.55 0-1-.45-1-1s.45-1 1-1s1 .45 1 1s-.45 1-1 1Z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h1
          className="text-center font-sans text-extraDarkPurple xxs:mt-[-18px] xxs:scale-75 xxs:pb-2 sm:mt-0 md:scale-90 md:pb-0 lg:pt-2"
        >
          Two-wheeler
        </h1>
      </div>
    </div>
    <div
      className="my-3 flex items-center justify-center"
    >
      <div
        className="flex justify-center"
      >
        <div
          className="px-2 xxs:scale-[70%] sm:w-80 sm:scale-[80%] md:scale-95 xl:w-96"
        >
          <div
            className="relative flex w-full flex-row items-stretch space-x-2 xxs:mb-2 xxs:pb-2 sm:mb-0 md:pt-4 lg:pb-0"
          >
            <input
              aria-describedby="button-addon2"
              aria-label="Search"
              className=" block w-[50%] min-w-0 flex-auto rounded-2xl border border-solid border-gray-300 bg-white bg-clip-padding px-3 py-1.5 pr-2 text-base font-normal text-gray-700 transition ease-in-out focus:border-purple focus:bg-white focus:text-gray-700 focus:outline-none"
              onChange={[Function]}
              placeholder="Min TL"
              type="search"
            />
            <input
              aria-describedby="button-addon2"
              aria-label="Search"
              className=" m-0 block w-[50%] min-w-0 flex-auto rounded-2xl border border-solid border-gray-300 bg-white bg-clip-padding px-3 py-1.5 text-base font-normal text-gray-700 transition ease-in-out focus:border-purple focus:bg-white focus:text-gray-700 focus:outline-none"
              onChange={[Function]}
              placeholder="Max TL"
              type="search"
            />
          </div>
          <div
            className="grid scale-90 grid-flow-col items-center justify-center gap-4 pt-4"
          >
            <button
              className="font-md flashing m-1 rounded-lg border-2 bg-extraDarkPurple py-1 px-5 text-[17px] text-white shadow-sm"
              onClick={[Function]}
            >
              filter
            </button>
            <button
              className="font-md flashing m-1 rounded-lg border-2 bg-extraDarkPurple py-1 px-5 text-[17px] text-white shadow-sm"
              onClick={[Function]}
            >
              reset
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
